cadc-storage-permissions/src/main/java/org/opencadc/inventory/permissions/Grant.java

```diff
@@ -0,0 +1,137 @@
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  $Revision: 4 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.inventory.permissions;
+
+import java.net.URI;
+import java.util.Date;
+import java.util.List;
+
+import org.opencadc.gms.GroupURI;
+import org.opencadc.inventory.InventoryUtil;
+
+/**
+ * Holds grant information about an object.
+ * 
+ * @author majorb
+ *
+ */
+public class Grant {
+
+    private URI artifactURI;
+    private Date releaseDate;
+    private List<GroupURI> readGroups;
+    private List<GroupURI> readWriteGroups;
+    
+    private boolean isPublic;
+    private Date evaluationDate;
+    
+    public Grant(URI artifactURI, Date releaseDate) {
+        this(artifactURI, releaseDate, null, null);
+    }
+    
+    public Grant(URI artifactURI, Date releaseDate, List<GroupURI> readGroups, List<GroupURI> readWriteGroups) {
+        InventoryUtil.assertNotNull(Grant.class, "artifactURI", artifactURI);
+        this.artifactURI = artifactURI;
+        if (releaseDate == null) {
+            isPublic = false;
+        } else {
+            isPublic = releaseDate.before(new Date());
+        }
+        this.releaseDate = releaseDate;
+        this.readGroups = readGroups;
+        this.readWriteGroups = readWriteGroups;
+        this.evaluationDate = new Date();
+    }
+    
+    public URI getArtifactURI() {
+        return artifactURI;
+    }
+    
+    public Date getReleaseDate() {
+        return releaseDate;
+    }
+
+    public List<GroupURI> getReadGroups() {
+        return readGroups;
+    }
+
+    public List<GroupURI> getReadWriteGroups() {
+        return readWriteGroups;
+    }
+    
+    public boolean isPublic() {
```

Comment:
remove isPublic method

Reply:
isPublic is a calculated attribute... remove the attribute then?

Reply:
nevermind.. answered above